rascal-lsp/src/main/rascal/util/LanguageServer.rsc

```diff
@@ -137,5 +144,111 @@ data InlayKind // this determines style
     | other(str name)
     ;
 
+// Decorations are a subset of css styling, in general you want to define
+// a standardStyle, an optionally defined overrides for dark and light themes
+data Decoration
+    = standardStyle(set[DecoratorOptions] options)
+    | darkStyleOverrides(set[DecoratorOptions] options)
+    | lightStyleOverrides(set[DecoratorOptions] options)
+    | isWholeLine(bool continueRenderingAfterLineEnding)
+    ;
+
+// limited set of operations are possible to be overrideable per range in the tree
+// mirrors https://code.visualstudio.com/api/references/vscode-api#DecorationInstanceRenderOptions
+data InstanceDecoration
+    = instanceStandard(set[InstanceDecorationOptions] options)
+    | instanceDarkOverrides(set[InstanceDecorationOptions] options)
+    | instanceLightOverrides(set[InstanceDecorationOptions] options)
+    ;
+
+data InstanceDecorationOptions
+    = instanceAfter(AttachableDecorationOptions tdaro)
+    | instanceBefore(AttachableDecorationOptions tdaro)
+    ;
+
+// css properties that decorators can change (in relation to the active theme and other decorations)
+// mirrors: https://code.visualstudio.com/api/references/vscode-api#ThemableDecorationRenderOptions
+data DecoratorOptions
+    // option to insert content after & before the decorated text
+    = after(AttachableDecorationOptions tdaro)
+    | before(AttachableDecorationOptions tdaro)
+
+    | backgroundColor(Color col)
+    | color(Color col)
+    | overviewRulerColor(Color col) // use transparant colors
+    | opacity(str opacity)
+
+    | border(str style)
+    | borderColor(Color col)
+    // subparts of border, useful for only overriding a single aspect of a parent definition
+    // try to prefer generic border defititions
+    | borderRadius(str style)
```

Comment:
what kind of styles are there and can they be enumerated? In that case use another `data` constructor